core/src/main/scala/slog/LoggingContext.scala

```diff
@@ -0,0 +1,15 @@
+package slog
+
+trait LoggingContext[F[_]] {
+  def use[T: StructureEncoder](key: String,
+                               value: T): LoggingContext.LoggingBuilder[F]
+}
+
+object LoggingContext {
+  def apply[F[_]](implicit ev: LoggingContext[F]): LoggingContext[F] = ev
+
+  trait LoggingBuilder[F[_]] {
+    def scoped[T](fv: F[T]): F[T]
```

Comment:
another option is `span` (as in tracing labraries). Or `use`, which reminds me of `Resource`, which also has this "enclosed block" meaning.